mats/cptypes.ms

```diff
@@ -880,97 +893,66 @@
   (not (cptypes-equivalent-expansion?
          '(lambda (x) (#2%exact? x))
          '(lambda (x) (#3%exact? x))))
-  (cptypes-equivalent-expansion?
-    '(lambda (x) (when (fixnum? x)
-                   (let ([y (add1 x)])
-                     (and (integer? y) (exact? y)))))
-    '(lambda (x) (when (fixnum? x)
-                   (let ([y (add1 x)])
-                     #t))))
   (not (cptypes-equivalent-expansion?
          '(lambda (x) (when (fixnum? x)
                         (fixnum? (add1 x))))
          '(lambda (x) (when (fixnum? x)
                         #t))))
-  (not (cptypes-equivalent-expansion?
-         '(lambda (x) (when (bignum? x)
-                        (bignum? (add1 x))))
-         '(lambda (x) (when (bignum? x)
-                        #t))))
-  (cptypes-equivalent-expansion?
-    '(lambda (x) (when (real? x)
-                   (real? (add1 x))))
-    '(lambda (x) (when (real? x)
-                   #t)))
   (cptypes-equivalent-expansion?
     '(lambda (x) (when (flonum? x)
-                   (flonum? (add1 x))))
+                   (add1 x)))
     '(lambda (x) (when (flonum? x)
-                   #t)))
+                   (#3%fl+ x 1.0))))
   (cptypes-equivalent-expansion?
     '(lambda (x) (when (flonum? x)
-                   (add1 x)))
+                   (1+ x)))
     '(lambda (x) (when (flonum? x)
                    (#3%fl+ x 1.0))))
-  (cptypes-equivalent-expansion?
-    '(lambda (x) (when (fixnum? x)
-                   (let ([y (sub1 x)])
-                     (and (integer? y) (exact? y)))))
-    '(lambda (x) (when (fixnum? x)
-                   (let ([y (sub1 x)])
-                     #t))))
   (not (cptypes-equivalent-expansion?
          '(lambda (x) (when (fixnum? x)
                         (fixnum? (sub1 x))))
          '(lambda (x) (when (fixnum? x)
                         #t))))
-  (not (cptypes-equivalent-expansion?
-         '(lambda (x) (when (bignum? x)
-                        (bignum? (sub1 x))))
-         '(lambda (x) (when (bignum? x)
-                        #t))))
```

Comment:
Should this test have been kept? I don't think it's covered by the `test-closed1` generalization, but I might be missing something else.

Reply:
No, because when x = (+ (most-positive-fixnum) 1), it's incorrect.

Reply:
I undeleted the test. It's a to ensure I don't make an easy mistake in the future. I don't know why I deleted it.

It's not included in `test-closed1`. Moreover, I was thinking about writing `test-not-closed1` to reduce duplication, but I decided that it was too weird. But Perhaps I'll add it in the future if there are too many cases of almost-closed functions.